internal/ir/xds.go

```diff
@@ -125,6 +141,63 @@ type HTTPListener struct {
 	Routes []*HTTPRoute
 	// IsHTTP2 is set if the upstream client as well as the downstream server are configured to serve HTTP2 traffic.
 	IsHTTP2 bool
+	// RequestAuthentication defines the schema for authenticating HTTP requests.
+	RequestAuthentication *RequestAuthentication
```

Comment:
shouldnt this be per HTTPRoute instead of per HTTPListener

Reply:
JWT is configured as an Envoy listener filter, specifically `envoy.filters.http.jwt_authn`. The request authen design [doc](https://github.com/envoyproxy/gateway/blob/main/docs/latest/design/request-authentication.md) provides examples on how the AuthentictionFilter<>HTTPRoutre resource get translated into an xDS config. The AuthenticationFilter spec gets translated into [jwt requirements](https://github.com/envoyproxy/gateway/pull/813/files#diff-b0330312d3f47f14bcdc8e1b589d84001a1969564c86ff59347bd56795db7ccaR177) and the `match` of the HTTPRoute rule with a `filter` of type AuthenticationFilter gets translated into [jwt matches](https://github.com/envoyproxy/gateway/pull/813/files#diff-b0330312d3f47f14bcdc8e1b589d84001a1969564c86ff59347bd56795db7ccaR175). The [JWT filter](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/jwt_authn/v3/config.proto#extension-envoy-filters-http-jwt-authn) processes the request before routing.

Reply:
would recommend looking into the route config here https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-route which allows you to the same thing using the `typed_per_filter_config` field within `Route`. This eliminates the need to duplicate the info in `httproute.spec.rules[].matches[]` since it is already applicable to the route

Reply:
@arkodg thanks for the review and pointer. Commit cf98e95 updates RequestAuthentication to be a field of HTTPRoute instead of HTTPListener.

Reply:
thanks for incorporating this, this will continue to keep the mapping of GAPI resources to xDS resources easy to maintain